pkg/storage/bucket/prefixed_bucket_client.go

```diff
@@ -23,7 +22,7 @@ func NewPrefixedBucketClient(bucket objstore.Bucket, prefix string) *PrefixedBuc
 }
 
 func (b *PrefixedBucketClient) fullName(name string) string {
-	return path.Join(b.prefix, name)
+	return b.prefix + objstore.DirDelim + name
```

Comment:
To keep the same exact logic we had in `UserBucketClient`. If name is "", the delimiter must be added anyway.

Reply:
I see, sorry I made this change.

Reply:
Based on my suggestion, so sorry about that :) 

I'm not sure empty name should be a valid choice though and I wonder how do object stores work in that case.

Reply:
> I'm not sure empty name should be a valid choice
It's used to list the root. Thanos does it.

Reply:
Ah right, I misunderstood and was thinking about empty prefix.